spanner/leaderboard/complete/src/main/java/com/google/codelabs/App.java

```diff
@@ -0,0 +1,353 @@
+ *   <li>Inserting data using a read-write transaction.
+ *   <li>Execute SQL queries over data, filtering and ordering by timestamp.
+ *   <li>Deleting a Cloud Spanner database.
+ * </ul>
+ */
+public class App {
+
+  static void create(DatabaseAdminClient dbAdminClient, DatabaseId db) {
+    OperationFuture<Database, CreateDatabaseMetadata> op =
+        dbAdminClient.createDatabase(
+            db.getInstanceId().getInstance(),
+            db.getDatabase(),
+            Arrays.asList(
+                "CREATE TABLE Players(\n"
+                    + "  PlayerId INT64 NOT NULL,\n"
+                    + "  PlayerName STRING(2048) NOT NULL\n"
+                    + ") PRIMARY KEY(PlayerId)",
+                "CREATE TABLE Scores(\n"
+                    + "  PlayerId INT64 NOT NULL,\n"
+                    + "  Score INT64 NOT NULL,\n"
+                    + "  Timestamp TIMESTAMP NOT NULL\n"
+                    + "  OPTIONS(allow_commit_timestamp=true)\n"
+                    + ") PRIMARY KEY(PlayerId, Timestamp),\n"
+                    + "INTERLEAVE IN PARENT Players ON DELETE NO ACTION"));
+    try {
+      // Initiate the request which returns an OperationFuture.
+      Database dbOperation = op.get();
+      System.out.println("Created database [" + dbOperation.getId() + "]");
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+
+  static void insert(DatabaseClient dbClient, String insertType) {
+    try {
+      insertType = insertType.toLowerCase();
+    } catch (Exception e) {
+      // Invalid input received, set inserttype to empty string.
+      insertType = "";
+    }
+    if (insertType.equals("players")) {
+      // Insert players.
+      insertPlayers(dbClient);
+    } else if (insertType.equals("scores")) {
+      // Insert scores.
+      insertScores(dbClient);
+    } else {
+      // Invalid input.
+      System.out.println("Invalid value for 'type of insert'. "
+          + "Specify a valid value: 'players' or 'scores'.");
+      System.exit(1);
+    }
+  }
+
+  static void insertPlayers(DatabaseClient dbClient) {
+    dbClient
+        .readWriteTransaction()
+        .run(
+            new TransactionCallable<Void>() {
+              @Override
+              public Void run(TransactionContext transaction) throws Exception {
+                // Get the number of players.
+                String sql =
+                    "SELECT Count(PlayerId) as PlayerCount FROM Players";
+                ResultSet resultSet = transaction.executeQuery(Statement.of(sql));
+                long numberOfPlayers = 0;
+                while (resultSet.next()) {
```

Comment:
Is this supposed to be a while? Isn't it just a single Result?

Reply:
Fixed.